### PR TITLE
Report aktualizr config to server

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
This relates to the ota-lite MR.

**Comment by doanac** [comment]
I need to test this in the morning.

**Comment by mike-sul** [comment]
@doanac Unformatted code

File: src/main.cc [hunk 1]
```diff
@@ -176,6 +176,8 @@ static int daemon_main(LiteClient& client, const bpo::variables_map& variables_m
     }
 
     if (firstLoop) {
+      client.reportAktualizrConfiguration();
```
Comment (mike-sul) [inline src/main.cc:179]: I am wondering if it can be invoked before `while (true)`/Update Loop?

File: src/helpers.cc [hunk 2]
```diff
@@ -467,6 +467,29 @@ std::pair<bool, Uptane::Target> LiteClient::downloadImage(const Uptane::Target&
...
+  std::stringstream conf_ss;
+  config.writeToStream(conf_ss);
+  const std::string conf_str = conf_ss.str();
+  const Hash new_hash = Hash::generate(Hash::Type::kSha256, conf_str);
+  std::string stored_hash;
+  if (!(storage->loadDeviceDataHash("configuration", &stored_hash) &&
+        new_hash == Hash(Hash::Type::kSha256, stored_hash))) {
+    LOG_DEBUG << "Reporting libaktualizr configuration";
+    const HttpResponse response = http_client->post(config.tls.server + "/system_info/config", "application/toml", conf_str);
+    if (response.isOk()) {
```
Comment (mike-sul) [inline src/helpers.cc:485]: maybe print some warning if !response.isOk?

File: src/helpers.cc [hunk 3]
```diff
@@ -467,6 +467,29 @@ std::pair<bool, Uptane::Target> LiteClient::downloadImage(const Uptane::Target&
...
+    return;
+  }
+
+  std::stringstream conf_ss;
+  config.writeToStream(conf_ss);
+  const std::string conf_str = conf_ss.str();
+  const Hash new_hash = Hash::generate(Hash::Type::kSha256, conf_str);
+  std::string stored_hash;
+  if (!(storage->loadDeviceDataHash("configuration", &stored_hash) &&
+        new_hash == Hash(Hash::Type::kSha256, stored_hash))) {
```
Comment (mike-sul) [inline src/helpers.cc:482]: `if (storage->loadDeviceDataHash("configuration", &stored_hash) && new_hash == Hash(Hash::Type::kSha256, stored_hash)) { LOG <<; return;}` 
IMO, the code will be slightly cleaner if we return if the config has not changed.

Comment (doanac) [reply src/helpers.cc:482]: Sure. I just took the code as-is from sotauptaneclient.cc.

File: src/helpers.cc (hunk 2 above)
Comment (doanac) [reply src/helpers.cc:485]: we aren't doing this for any of the "report" methods. I'll add a 2nd commit to this PR that does that.

File: src/helpers.cc (hunk 3 above)
Comment (doanac) [reply src/helpers.cc:482]: So i've updated the current first commit of this PR to just do like our other report* methods do. I can then do follow up commits to clean this code.

**Review by mike-sul — APPROVED** [review]

File: src/helpers.cc (hunk 3 above)
Comment (doanac) [reply src/helpers.cc:482]: I've tried to clean things up with the latest push. Let me know what you think

**Review by mike-sul — APPROVED** [review]

**Pull request merged**